mantidproject/mantid
# Allow user to specify max iterations and make partial spectra fitting more flexible.

**Description of work.**

There are two major improvements for FitPeaks.

1.  An optional output TableWorkspace is added for showing fit function parameters' uncertainties.
2. Both output TableWorksapces, for parameters and uncertainties, will have same number of rows as the number the spectra to fit.

**To test:**

1. Check new unit tests
2. Run FitPeaks on a multiple spectra workspace to fit peaks in a part of spectra.  Check both of the output TableWorkspace for fitted parameter value and uncertainties.

Fixes #22860.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
